iOS/DuckDuckGo/AIChat/AIChatViewControllerManager.swift

```diff
@@ -223,7 +277,11 @@ extension AIChatViewControllerManager: AIChatViewControllerDelegate {
 
 extension AIChatViewControllerManager: RoundedPageSheetContainerViewControllerDelegate {
     func roundedPageSheetContainerViewControllerDidDisappear(_ controller: RoundedPageSheetContainerViewController) {
-        cleanUpUserContent()
+        guard isKeepSessionEnabled == false else { return }
+
+        Task { @MainActor in
```

Comment:
I think it's better to mark `cleanUpSession` or callees inside of it as `@MainActor` instead of doing this for the task closure.